enterprise/backend/src/metabase_enterprise/sandbox/models/params/field_values.clj

```diff
@@ -29,18 +74,18 @@
   "Returns a hash-key for linked-filter FieldValues if the field is sandboxed, otherwise fallback to the OSS impl."
   :feature :sandboxes
   [field-id constraints]
-  (if (field-is-sandboxed? (db/select-one Field :id field-id))
-    (str (hash [api/*current-user-id*
-                @api/*current-user-permissions-set*
-                field-id
-                constraints]))
-    (field-values/default-hash-key-for-linked-filters field-id constraints)))
+  (let [field (db/select-one Field :id field-id)]
+    (if (field-is-sandboxed? field)
+      (str (hash (concat [field-id
+                          constraints]
+                         (field->gtap-attributes-for-current-user field))))
```

Comment:
Do you think it would make it cleaner to grab `api/*current-user*` and pass that down through the functions below rather than having those functions reach out to this state? Might make testing easier to not have to bind these values and just take a parameter?

Reply:
Looking at it a little more, I wonder if we could move the access to `api/*current-user*` closer to the api layer. If its too much we can leave it as is but I'm a big fan of api routes grabbing that value and then other functions taking that as a parameter.

Reply:
I tried it but it was quite awkward because there are some interconnect of this ns with the chain-filter ns. 
If we do that we would need to update the chain-filter ns to pass the `current-user` too.